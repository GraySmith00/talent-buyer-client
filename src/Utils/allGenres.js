export const allGenres = [
  'big room',
  'dance pop',
  'edm',
  'electro house',
  'pop',
  'progressive electro house',
  'tropical house',
  'house',
  'minimal techno',
  'schranz',
  'tech house',
  'techno',
  'electropop',
  'indie r&b',
  'indietronica',
  'deep big room',
  'dutch house',
  'progressive house',
  'folk',
  'folk-pop',
  'lilith',
  'mellow gold',
  'new wave pop',
  'pop rock',
  'scottish new wave',
  'soft rock',
  'alternative dance',
  'australian dance',
  'filter house',
  'big beat',
  'disco house',
  'electronic',
  'new rave',
  'uk garage',
  'vocal house',
  'aussietronica',
  'indie poptimism',
  'vapor soul',
  'progressive uplifting trance',
  'uplifting trance',
  'art pop',
  'chamber pop',
  'experimental pop',
  'icelandic pop',
  'metropopolis',
  'permanent wave',
  'trip hop',
  'afro house',
  'deep house',
  'kwaito house',
  'bass trap',
  'brostep',
  'dubstep',
  'electronic trap',
  'german techno',
  'microhouse',
  'contemporary country',
  'country',
  'country road',
  'modern country rock',
  'redneck',
  'brazilian electronica',
  'deep groove house',
  'melbourne bounce',
  'tropical pop edm',
  'deep disco house',
  'deep melodic euro house',
  'minimal tech house',
  'tribal house',
  'chamber psych',
  'float house',
  'shiver pop',
  'downtempo',
  'hip hop',
  'turntablism',
  'adult standards',
  'cabaret',
  'christmas',
  'easy listening',
  'lounge',
  'swing',
  'vocal jazz',
  'catstep',
  'complextro',
  'filthstep',
  'progressive trance',
  'trance',
  'bass music',
  'future garage',
  'dance-punk',
  'electroclash',
  'neo-synthpop',
  'nu jazz',
  'nu skool breaks',
  'australian hip hop',
  'deep australian indie',
  'escape room',
  'chillwave',
  'indie psych-rock',
  'nu disco',
  'shimmer pop',
  'vaporwave',
  'wonky',
  'acid house',
  'chicago house',
  'hip house',
  'laboratorio',
  'new romantic',
  'new wave',
  'vintage french electronic',
  'deep minimal techno',
  'breakbeat',
  'hardcore techno',
  'norwegian hip hop',
  'norwegian pop',
  'vapor twitch',
  'deep tropical house',
  'ccm',
  'christian alternative rock',
  'christian music',
  'christian rock',
  'world worship',
  'worship',
  'indie pop',
  'indie rock',
  'modern rock',
  'swedish electropop',
  'swedish indie pop',
  'swedish synthpop',
  'etherpop',
  'gauze pop',
  'drum and bass',
  'liquid funk',
  'intelligent dance music',
  'ambient',
  'art rock',
  'compositional ambient',
  'dream pop',
  'drone',
  'experimental',
  'experimental rock',
  'fluxwork',
  'fourth world',
  'freak folk',
  'hauntology',
  'mandible',
  'neo-psychedelic',
  'noise pop',
  'noise rock',
  'outsider house',
  'post-rock',
  'glitch hop',
  'banda',
  'grupera',
  'norteno',
  'ranchera',
  'regional mexican',
  'dance rock',
  'europop',
  'synthpop',
  'post-teen pop',
  'strut',
  'acid techno',
  'detroit techno',
  'dub techno',
  'modern alternative rock',
  'chillstep',
  'tracestep',
  'hardstyle',
  'electro',
  'belgian indie',
  'mashup',
  'neurofunk',
  'acid jazz',
  'danish electro-pop',
  'finnish edm',
  'deep euro house',
  'vapor pop',
  'indie anthem-folk',
  'indie electro-pop',
  'viral pop',
  'indian indie',
  'pop house',
  'pop rap',
  'alternative rock',
  'garage rock',
  'rock',
  'electro swing',
  'indie dream pop',
  'country gospel',
  'country rock',
  'cowboy western',
  'nashville sound',
  'outlaw country',
  'traditional country',
  'northern irish indie',
  'latvian pop',
  'dutch hip hop',
  'dark jazz',
  'grave wave',
  'destroy techno',
  'rap',
  'underground hip hop',
  'chillhop',
  'jazz boom bap',
  'la indie',
  'stomp and holler',
  'r&b',
  'british jazz',
  'broken beat',
  'indie jazz',
  'moombahton',
  'zapstep',
  'wave',
  'candy pop',
  'scottish hip hop',
  'australian alternative rock',
  'japanese classical performance',
  'violin',
  'australian pop',
  'eurodance',
  'neo soul',
  'british contemporary classical',
  'british soundtrack',
  'soundtrack',
  'swedish pop',
  'emo rap',
  'krautrock',
  'austindie',
  'gangster rap',
  'southern hip hop',
  'trap music',
  'abstract beats',
  'arab alternative',
  'brooklyn indie',
  'dreamo',
  'french indietronica',
  'jam band',
  'pop punk',
  'alternative hip hop',
  'east coast hip hop',
  'hardcore hip hop',
  'melancholia',
  'uk contemporary r&b',
  'britpop',
  'madchester',
  'post-punk',
  'uk post-punk',
  'athens indie',
  'indie emo'
];
