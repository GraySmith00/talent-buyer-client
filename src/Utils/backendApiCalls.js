import axios from 'axios';
import { artistsByAgency } from './artistsByAgency';
import { getArtistInfo } from './songKickApiCalls';

export const signUpPostRequest = async user => {
  const url = `${process.env.REACT_APP_API}/signup`;
  const response = await axios.post(url, user);
  return response.data;
};

export const logInPostRequest = async userCreds => {
  const url = `${process.env.REACT_APP_API}/login`;
  const response = await axios.post(url, userCreds);
  if (response.status === 200) {
    return response;
  }
};

export const venuePostRequest = async venue => {
  const url = `${process.env.REACT_APP_API}/api/v1/venues`;
  const response = await axios.post(url, venue);
  const data = response.data;
  return data;
};

export const getAllUserVunues = async userId => {
  const url = `${process.env.REACT_APP_API}/api/v1/buyers/${userId}/venues`;
  const response = await axios.get(url);
  const data = response.data;
  return data;
};

export const artistPostRequest = async artist => {
  const url = `${process.env.REACT_APP_API}/api/v1/artists`;
  await fetch(url, {
    method: 'POST',
    body: JSON.stringify(artist),
    headers: {
      'Content-Type': 'application/json'
    }
  });
};

export const addWmeArtistsToBackend = () => {
  artistsByAgency.WME.map(async artistName => {
    const artistInfo = await getArtistInfo(artistName, 'WME');
    if (artistInfo) {
      await artistPostRequest(artistInfo);
    }
  });
};

export const addCaaArtistsToBackend = () => {
  artistsByAgency.CAA.map(async artistName => {
    const artistInfo = await getArtistInfo(artistName, 'CAA');
    if (artistInfo) {
      await artistPostRequest(artistInfo);
    }
  });
};

export const getAllArtists = async () => {
  const url = `${process.env.REACT_APP_API}/api/v1/artists`;
  const artists = await axios.get(url);
  return artists.data;
};

export const getArtist = async id => {
  const url = `${process.env.REACT_APP_API}/api/v1/artists/${id}`;
  const artist = await axios.get(url);
  return artist.data;
};

export const offerPostRequest = async offer => {
  const url = `${process.env.REACT_APP_API}/api/v1/offers`;
  const savedOffer = await axios.post(url, offer);
  return savedOffer.data;
};

export const getAllUserOffers = async userId => {
  const url = `${process.env.REACT_APP_API}/api/v1/buyers/${userId}/offers`;
  const response = await axios.get(url);
  return response.data;
};

export const offerPutRequest = async offer => {
  const url = `${process.env.REACT_APP_API}/api/v1/buyers/${
    offer.buyer_id
  }/offers/${offer.id}`;
  const response = await axios.put(url, offer);
  return response.data;
};

export const watchlistPostRequest = async artistId => {
  const url = `${process.env.REACT_APP_API}/api/v1/favorite_artists`;
  const response = await axios.post(url, { artist_id: artistId });
  return response;
};

export const watchlistDeleteRequest = async artistId => {
  const url = `${
    process.env.REACT_APP_API
  }/api/v1/favorite_artists/${artistId}`;
  const response = await axios.delete(url);
  return response;
};

export const getUserWatchlist = async () => {
  const url = `${process.env.REACT_APP_API}/api/v1/favorite_artists`;
  const response = await axios.get(url);
  return response;
};
