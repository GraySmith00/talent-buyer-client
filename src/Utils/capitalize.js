export const capitalize = string => {
  return string
    .split(' ')
    .map(word => word.slice(0, 1).toUpperCase() + word.slice(1))
    .join(' ');
};

export const capitalizeArtists = searchResults => {
  if (searchResults.length) {
    const results = searchResults.map(artist => {
      let cleanedArtists = [];
      const split = artist.split(' ');
      split.forEach(name => {
        cleanedArtists.push(name[0].toUpperCase() + name.slice(1));
      });
      return cleanedArtists.join(' ');
    });
    return results;
  }
};
