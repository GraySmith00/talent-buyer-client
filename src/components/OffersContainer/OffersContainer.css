.offers-container .table-headings {
  display: grid;
  grid-template-columns: repeat(4, 10fr);
  border-bottom: 1px solid #fff;
  align-items: center;
  padding-left: 2rem;
  margin: 1rem 0 0 0; }
  .offers-container .table-headings h3 {
    margin-bottom: 0; }

.offers-container .rows {
  height: 140px;
  overflow: scroll;
  background-color: #32333e; }

.offers-container .table-row {
  display: grid;
  grid-template-columns: repeat(4, 10fr);
  align-items: center;
  justify-content: center;
  margin-left: 2rem;
  font-weight: 300; }
