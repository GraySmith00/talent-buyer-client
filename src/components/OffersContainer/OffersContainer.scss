.offers-container {
  height: 140px;
  overflow: scroll;
  background-color: #32333e;
}

.table-row {
  display: grid;
  grid-template-columns: repeat(3, 10fr);
  align-items: center;
  justify-content: center;
  margin-left: 2rem;
  font-weight: 300;
}
