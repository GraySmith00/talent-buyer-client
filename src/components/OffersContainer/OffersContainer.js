import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { capitalize } from '../../Utils/capitalize';
import './OffersContainer.css';

export class OffersContainer extends Component {
  formatDate = date => {
    let day = date.getDate();
    if (day < 10) day = `0${day}`;
    const month = date.getMonth() + 1;
    const year = date.getFullYear();
    return `${year}-${month}-${day}`;
  };

  handleEditOffer = id => {
    this.props.openEditOfferModal(id);
  };

  handleViewOffer = id => {
    this.props.openViewOfferModal(id);
  };

  render() {
    const { offers, date } = this.props;
    const formattedDate = this.formatDate(date);

    const filteredOffers = offers.filter(offer => offer.date === formattedDate);

    let displayOffers;

    if (filteredOffers.length) {
      displayOffers = filteredOffers.map(offer => (
        <div className="table-row" key={offer.id}>
          <h3>{offer.artist_name}</h3>
          <h3>{capitalize(offer.status)}</h3>
          <button
            className="view-offer-button"
            onClick={() => this.handleEditOffer(offer.id)}
          >
            Edit Offer
          </button>
          <button
            className="view-offer-button"
            onClick={() => this.handleViewOffer(offer.id)}
          >
            View Offer
          </button>
        </div>
      ));
    }

    return <div className="offers-container">{displayOffers}</div>;
  }
}

OffersContainer.propTypes = {
  offers: PropTypes.array.isRequired,
  date: PropTypes.object.isRequired,
  openEditOfferModal: PropTypes.func.isRequired,
  openViewOfferModal: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
  offers: state.offers
});

export default connect(mapStateToProps)(OffersContainer);
