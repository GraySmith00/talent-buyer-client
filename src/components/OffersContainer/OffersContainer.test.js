import React from 'react';
import { shallow } from 'enzyme';
import { mockOffers } from '../../Utils/mockData';
import { OffersContainer } from './OffersContainer';

describe('OffersContainer component', () => {
  let wrapper;
  let mockDate;
  let mockOpenEditOfferModal;

  beforeEach(() => {
    mockDate = new Date(2019, 11, 10);
    mockOpenEditOfferModal = jest.fn();

    wrapper = shallow(
      <OffersContainer
        date={mockDate}
        offers={mockOffers}
        openEditOfferModal={mockOpenEditOfferModal}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
