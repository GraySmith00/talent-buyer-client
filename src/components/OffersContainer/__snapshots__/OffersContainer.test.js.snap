// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersContainer component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OffersContainer
    date={2019-12-10T07:00:00.000Z}
    offers={
        Array [
            Object {
              "age_range": "16+",
              "artist_id": 19,
              "artist_name": "Calvin Harris",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-29",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 5,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 19,
              "artist_name": "Calvin Harris",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-29",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 6,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 19,
              "artist_name": "Calvin Harris",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-29",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 7,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 4,
              "artist_name": "Alesso",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-29",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 100032,
              "id": 8,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 7,
              "artist_name": "Afrojack",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-29",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 9,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 7,
              "artist_name": "Afrojack",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-29",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 10,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 99,
              "artist_name": "Savoy",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-29",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 11,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 4,
              "artist_name": "Alesso",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-29",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 12,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 22,
              "artist_name": "Chris Lake",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-29",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 13,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 19,
              "artist_name": "Calvin Harris",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-30",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 14,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 7,
              "artist_name": "Afrojack",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-29",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 15,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 4,
              "artist_name": "Alesso",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-11-24",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 16,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 16,
              "artist_name": "Borgore",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-10-29",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 17,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 4,
              "artist_name": "Alesso",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-11-16",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 18,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 16,
              "artist_name": "Borgore",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-11-01",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 19,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 4,
              "artist_name": "Alesso",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-11-01",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 20,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 7,
              "artist_name": "Afrojack",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-11-08",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 21,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
            Object {
              "age_range": "16+",
              "artist_id": 18,
              "artist_name": "Booka Shade",
              "bonuses": "+$1k at 600 sold",
              "buyer_id": 3,
              "date": "2018-11-02",
              "door_times": "9pm-12am",
              "gross_potential": 8500,
              "guarantee": 5000,
              "id": 22,
              "merch_split": "80/20 artist/venue",
              "radius_clause": "120mi +/- 60days",
              "status": "pending",
              "total_expenses": 2000,
              "venue_id": 3,
            },
          ]
    }
    openEditOfferModal={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "offers-container",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "offers-container",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
